### PR TITLE
Adapt workspace download config

### LINKED ISSUE #6351: Allow admin users to download workspaces
We once had a "Download workspace content" button on the admin dashboard which was quite handy when trying to analyze abuse. Would be nice to have it again!

missing-button

- [x] Add download button to admin dashboard
- [ ] Adjust the [workspace download route config](https://github.com/gitpod-io/gitpod/issues/6351#issuecomment-982373095)

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Adapts the route to also work with other domains. 

## Related Issue(s)

Fixes #6351

## Release Notes

```release-note
Workspace download route config set to work with other domains.
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#6351*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/proxy/plugins/workspacedownload/workspace_download.go [hunk 1]
```diff
@@ -75,12 +75,31 @@ func (m Download) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyht
...
-		return fmt.Errorf("Server error: cannot obtain redirect URL")
+		return fmt.Errorf("server error: cannot obtain workspace download redirect URL")
 	}
+	upstreamURL := string(upstreamURLBytes)
 
-	repl.Set("http."+workspaceDownloadModule+"_url", string(redirectURL))
+	// perform the upstream request here
+	resp, err = http.Get(upstreamURL)
+	if err != nil {
+		caddy.Log().Sugar().Errorf("error starting download of prebuild log for %v: %v", upstreamURL, err)
```
Comment (geropl) [inline components/proxy/plugins/workspacedownload/workspace_download.go:87]: `prebuild log` -> `workspace`

File: components/proxy/plugins/workspacedownload/workspace_download.go [hunk 2]
```diff
@@ -75,12 +75,31 @@ func (m Download) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyht
...
+		return caddyhttp.Error(http.StatusInternalServerError, fmt.Errorf("unexpected error downloading prebuild log"))
```
Comment (geropl) [inline components/proxy/plugins/workspacedownload/workspace_download.go:88]: `prebuild log` -> `workspace`

**Comment by geropl** [comment]
@laushinka Just tested, works nicely for these case:
 - [x] own workspace (from `/workspaces`)
 - [x] as admin from `/admin`

It would be nice if you had a look at the comments/log statements mentioned above, then this is ready to :ship: ! Holding because of this.

/lgtm
/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 41012385439c999e8daf1aff07421b8cfd6a37d3</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6988#issuecomment-982760967" title="LGTM">geropl</a>*

Associated issue: *#6351*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/proxy/plugins/workspacedownload/workspace_download.go [hunk 3]
```diff
@@ -75,12 +75,31 @@ func (m Download) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyht
 		return fmt.Errorf("Bad Request: /workspace-download/get returned with code %v", resp.StatusCode)
 	}
 
-	redirectURL, err := io.ReadAll(resp.Body)
+	upstreamURLBytes, err := io.ReadAll(resp.Body)
 	if err != nil {
-		return fmt.Errorf("Server error: cannot obtain redirect URL")
+		return fmt.Errorf("server error: cannot obtain workspace download redirect URL")
```
[the remark this answers is not in the document]
Comment (laushinka) [reply components/proxy/plugins/workspacedownload/workspace_download.go:80]: Also just noticed that the cases are quite inconsistent in this file 😅

**Comment by roboquat** [comment]
New changes are detected. LGTM label has been removed.

**Comment by laushinka** [comment]
@geropl Pushed and need dat lgtm again 🤗

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 14bbc00c4e54a2f829d7c828f42843fac0161d05</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6988#issuecomment-983428582" title="LGTM">geropl</a>*

Associated issue: *#6351*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**